plugins/source/aws/resources/services/wellarchitected/lens_review_improvements.go

```diff
@@ -0,0 +1,86 @@
+package wellarchitected
+
+import (
+	"context"
+
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/aws/aws-sdk-go-v2/service/wellarchitected"
+	"github.com/aws/aws-sdk-go-v2/service/wellarchitected/types"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/plugin-sdk/v3/schema"
+	"github.com/cloudquery/plugin-sdk/v3/transformers"
+)
+
+func lensReviewImprovements() *schema.Table {
+	name := "aws_wellarchitected_lens_review_improvements"
+	return &schema.Table{
+		Name:        name,
+		Description: `https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_ImprovementSummary.html`,
+		Transform: transformers.TransformWithStruct(new(types.ImprovementSummary),
+			transformers.WithPrimaryKeys("PillarId", "QuestionId"),
+			transformers.WithUnwrapAllEmbeddedStructs(),
+		),
+		Multiplex: client.ServiceAccountRegionMultiplexer(name, "wellarchitected"),
+		Resolver:  fetchLensReviewImprovements,
+		Columns: schema.ColumnList{
+			client.DefaultAccountIDColumn(true),
+			client.DefaultRegionColumn(true),
+			{
+				Name:       "workload_id",
+				Type:       arrow.BinaryTypes.String,
+				Resolver:   schema.ParentColumnResolver("workload_id"),
+				PrimaryKey: true,
+			},
+			{
+				Name:       "milestone_number",
+				Type:       arrow.PrimitiveTypes.Int64,
+				Resolver:   schema.ParentColumnResolver("milestone_number"),
+				PrimaryKey: true,
+			},
+			{
+				Name:       "lens_alias",
+				Type:       arrow.BinaryTypes.String,
+				Resolver:   schema.ParentColumnResolver("lens_alias"),
+				PrimaryKey: true,
+			},
+		},
+	}
+}
+
+func fetchLensReviewImprovements(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	cl := meta.(*client.Client)
+	service := cl.Services().Wellarchitected
+	milestoneNumber := int32(parent.Get("milestone_number").Get().(int64))
+	workloadID := parent.Get("workload_id").String()
+	lensAlias := parent.Get("lens_alias").String()
```

Comment:
These should go after the casting of the parent to `types.LensReview` so that if we end early we won't have done the casting for nothing

Reply:
4c101cf80c537e51103a44e05febc1869fc3d7ff